NavigationBarAssignment/NavigationBarAssignment/ListProductsViewController.m

```diff
@@ -0,0 +1,132 @@
+    UIBarButtonItem *barButtonItemAdd = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAdd target:self action:@selector(addProducts)];
+    self.navigationController.topViewController.navigationItem.rightBarButtonItem = barButtonItemAdd;
+}
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+- (void)viewDidAppear:(BOOL)animated{
+    [self displayViewsList];
+}
+-(void) viewWillDisappear:(BOOL)animated{
+    [viewNoItem removeFromSuperview];
+}
+/** Navigates to screen add product details
+ */
+-(void)addProducts {
+    AddProductViewController *addProductViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"AddProductViewController"];
+    [self.navigationController pushViewController:addProductViewController animated:YES];
+}
+///** Displays view list displaying product details if available else returns no product available
+// */
+-(void)displayViewsList{
+    AppDelegate *appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
+    int arrayLength = [appDelegate.arrayProducts count];
+    if(arrayLength == 0){
+        [self viewsListForNoItem];
+   }
+    else{
+        [self viewsListForItems];
+    }
+}
+///** Displays that no product is available
+// */
+-(void)viewsListForNoItem{
+    viewNoItem = [[UIView alloc] initWithFrame: CGRectMake(10, 100, 300, 50)];
+    viewNoItem.backgroundColor = [UIColor cyanColor];
+    [self.view addSubview:viewNoItem];
+    UILabel *labelNoItem = [[UILabel alloc] initWithFrame: CGRectMake(10,10,200,20)];
+    labelNoItem.backgroundColor = [UIColor clearColor];
+    labelNoItem.text = @"No Item available";
+    [viewNoItem  addSubview:labelNoItem];
+}
+///** Displays view list displaying product list
+// */
+-(void)viewsListForItems{
+    AppDelegate *appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
+    int arrayLength=[appDelegate.arrayProducts count];
+    int yvalue = 20;
+    UIScrollView *scrollViewListProducts = [[UIScrollView alloc] initWithFrame:CGRectMake(10, 100, self.view.frame.size.width, self.view.frame.size.height)];
+    for(int i=0;i<arrayLength;i++){
+        UIView *viewEachProduct = [self createViewWithYValue:yvalue];
+        [scrollViewListProducts addSubview:viewEachProduct];
+        UIButton *buttonEachProduct = [self createButtonWithYValue:yvalue];
+        [buttonEachProduct setTag:i];
+        [scrollViewListProducts addSubview:buttonEachProduct];
+        UILabel *labelProductName = [self createLabelProductName:[appDelegate.arrayProducts[i] productName]];
+        [viewEachProduct  addSubview:labelProductName];
+       UILabel *labelProductPrice = [self createLabelProductprice:[NSString stringWithFormat:@"%0.2f", [appDelegate.arrayProducts[i] productPrice]]];
+       [viewEachProduct  addSubview:labelProductPrice];
+        yvalue += 70;
+    }
+    [self.view addSubview:scrollViewListProducts];
+}
```

Comment:
it should be more modular code, its name should be like "reloadProductList"
it should have helper method like addViewWithProduct: yOffset which will return fully designed viewProductList just to add on self.view